mqemitter.js

```diff
@@ -116,9 +116,14 @@ MQEmitter.prototype.close = function close (cb) {
 MQEmitter.prototype._do = function (message, callback) {
   this._doing = true
   const matches = this._matcher.match(message.topic)
+  const matchesCopy = new Array(matches.length)
+  for (let i = 0, len = matches.length; i < len; i++) {
+    matchesCopy[i] = matches[i]
+  }
```

Comment:
This will add too much overhead. Let's do the reverse: whenever `removeListener()` is called, a new array is allocated.

Reply:
ok seems fair,
Im not sure how we'd implement that, because the array with the matched handlers is given to fastparallel in `_do()`
and the `removeListener()` is called within a matched handler that was called by fastparallel?

Reply:
Actually there is a better way: move the removal inside a setImmediate.